ang/civicase/case/actions/services/email-case-action.service.js

```diff
@@ -47,12 +48,51 @@
       return model.deferObject.promise;
     };
 
+    /**
+     * @param {string|number[]} caseRoleIds list of case roles ids
```

Comment:
why do we pass two different parameter types? we should enforce a single type.

Reply:
I am not sure if this is possible at the moment. Because at different places of the application we have different kinds of data type. For example, case id can be both string or number, and this problem is present throughout the application both in FE and BE. I think it will be a huge thing now to change all of this. Thoughts?

Reply:
let's avoid this in the future and remove any such pattern whenever we see it. A function that accepts multiple types is probably doing extra steps to normalise the data to a particular type and that takes away from understanding the real purpose of the function.

In this case it could be fine, but I can see use-cases  of this same function where things would not work as expected.

`['client']` works, and `'client'` works as well, but `'notaclient'` works too and it should not.